src/services/dbtProjectService.ts

```diff
@@ -0,0 +1,50 @@
+import { Uri, window } from "vscode";
+import { DBTProject } from "../manifest/dbtProject";
+import { DBTProjectContainer } from "../manifest/dbtProjectContainer";
+import { ManifestCacheProjectAddedEvent } from "../manifest/event/manifestCacheChangedEvent";
+import { provideSingleton } from "../utils";
+
+@provideSingleton(DbtProjectService)
+export class DbtProjectService {
```

Comment:
Naming can be improved. This class is more of a wrapper around the manifest, to make it easier to query the manifest. Also why you don't let it manage the eventMap updates as well? That way we can reuse it in a lot more places.

Just search for `Map<string, ManifestCacheProjectAddedEvent>` and you see what I mean.

Reply:
https://github.com/AltimateAI/vscode-dbt-power-user/pull/910/commits/3de02a01bbc569f55351cd9d55964b6aac698e8f Modified and cleaned up some existing code.

Reply:
Can you also rename this to `QueryManifestService` (also the filename)?